frontend/test/test_oqd/oqd/test_oqd_database_managers.py

```diff
@@ -0,0 +1,507 @@
+            stage = "Loading"
+            process = "Ablation"
+            equation = ""
+            value = 532
+            unit = "nm"
+            """
+        )
+
+        properties = OQDDeviceDatabase.from_toml(toml_document)
+        assert properties.parameters
+
+        assert properties.parameters["N_load"]
+        assert properties.parameters["N_load"].name == "N_load"
+        assert properties.parameters["N_load"].description == "Number of ions"
+        assert properties.parameters["N_load"].stage == "Loading"
+        assert properties.parameters["N_load"].process == "Ablation"
+        assert properties.parameters["N_load"].equation == ""
+        assert properties.parameters["N_load"].value == 1
+        assert properties.parameters["N_load"].unit == ""
+
+        assert properties.parameters["w_ablation"]
+        assert properties.parameters["w_ablation"].name == "w_ablation"
+        assert properties.parameters["w_ablation"].description == "Ablation laser frequency"
+        assert properties.parameters["w_ablation"].stage == "Loading"
+        assert properties.parameters["w_ablation"].process == "Ablation"
+        assert properties.parameters["w_ablation"].equation == ""
+        assert properties.parameters["w_ablation"].value == 532
+        assert properties.parameters["w_ablation"].unit == "nm"
+
+    def test_from_toml_invalid(self):
+        """
+        Tests that the OQDDeviceDatabase.from_toml method raises a ValueError when the TOML document
+        is invalid.
+        """
+        with pytest.raises(ValueError, match="Failed to load TOML document"):
+            OQDDeviceDatabase.from_toml("Invalid TOML document")
+
+
+class TestOQDQubitDatabase:
+    """Test suite for the OQDQubitDatabase class."""
+
+    def test_from_toml_production(self):
+        """
+        Tests that the OQDQubitDatabase.from_toml method can load the qubit parameters from the
+        oqd_qubit_parameters.toml file used in production.
+
+        Here, we only test that we can load the file; we do not check any values since they are
+        subject to change in the production TOML file.
+        """
+        qubit_database = OQDQubitDatabase.from_toml(OQD_SRC_DIR / "oqd_qubit_parameters.toml")
+        assert qubit_database
+        assert qubit_database.ion_parameters
+        assert qubit_database.phonon_parameters
+
+    def test_from_toml(self):
+        """
+        Tests that the OQDQubitDatabase.from_toml method can load the qubit parameters from a sample
+        oqd_qubit_parameters.toml file.
+        """
+        qubit_database = OQDQubitDatabase.from_toml(OQD_TEST_DIR / "oqd_qubit_parameters.toml")
+        assert qubit_database
+
+        # Check ion parameters
+        assert qubit_database.ion_parameters
+        assert qubit_database.ion_parameters["Yb171"]
+
+        ion_params_yb171 = qubit_database.ion_parameters["Yb171"]
+        assert ion_params_yb171.mass == 171
```

Comment:
I guess here since the values are actually checked, the first two "existence checks" are not necessary?

Reply:
(alongside all other similar places)

Reply:
Strictly speaking, no they're not necessary. However, I'd like to keep this as is since with the many layers of nested dictionaries in the TOML data, this should make it easier to debug if a problem arises. If 

```python
assert qubit_database.ion_parameters
```

passes but

```python
assert qubit_database.ion_parameters["Yb171"]
```

fails, then we know that it was able to parse some ion parameters, but not those for `Yb171`.

Reply:
No objections!